providers/nitro/nitro-helper/src/key_utils.rs

```diff
@@ -1,45 +1,99 @@
-use bytes::Bytes;
+use crate::shared::AwsCredentials;
+use aws_sdk_kms::config::Config;
+use aws_sdk_kms::Credentials as KmsCredentials;
+use aws_sdk_kms::{Client as KmsClient, Region};
 use ed25519_dalek::{Keypair, PublicKey};
 use rand_core::OsRng;
-use rusoto_core::{region::Region, HttpClient};
-use rusoto_credential::InstanceMetadataProvider;
-use rusoto_kms::{EncryptRequest, Kms, KmsClient};
-use std::str::FromStr;
 use std::{fs::OpenOptions, io::Write, os::unix::fs::OpenOptionsExt, path::Path};
+use tokio::runtime::Runtime;
+
+// TODO: use aws-rust-sdk after the issue fixed
+// https://github.com/awslabs/aws-sdk-rust/issues/97
+pub(crate) mod credential {
+    use crate::shared::AwsCredentials;
+    use chrono::{DateTime, Utc};
+    use serde::Deserialize;
+    use std::collections::BTreeMap;
+
+    const AWS_CREDENTIALS_PROVIDER_IP: &str = "169.254.169.254";
+    const AWS_CREDENTIALS_PROVIDER_PATH: &str = "latest/meta-data/iam/security-credentials";
+
+    #[derive(Clone, Debug, Deserialize, Default)]
+    pub struct AwsCredentialsResponse {
+        #[serde(rename = "AccessKeyId")]
+        key: String,
+        #[serde(rename = "SecretAccessKey")]
+        secret: String,
+        #[serde(rename = "SessionToken", alias = "Token")]
+        token: Option<String>,
```

Comment:
would it be possible to use [SecretString](https://docs.rs/secrecy/0.7.0/secrecy/type.SecretString.html) for these, so they don't accidentally leak in the logs etc.?

Reply:
also zero-out them on drop as in https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/aws-auth/src/lib.rs#L27

Reply:
the SecretString already zero-out when drop